test/README.md

```diff
@@ -1,95 +1,296 @@
-
-_Note: You will need to increase both `fs.inotify.max_user_instances` and `fs.inotify.max_user_watches`_
+# End-to-end tests
+
+## Test code structure
+
+There are three directories under `test/`:
+
+1. `test/Microsoft.Azure.Devices.Edge.Test/`: The actual tests live here, as do "fixtures" (setup and tear-down code that runs before/after/between tests) and the `Context` class (exposes the test arguments to tests).
+2. `test/Microsoft.Azure.Devices.Edge.Test.Common/`: The helper library tests use to do interesting things--like deploy modules or wait for a message to arrive in IoT Hub--lives here.
+3. `test/modules`: All test modules live here. The common pattern for an end-to-end test is to deploy some modules that run through a scenario (i.e., the thing you actually want to test, like sending messages between modules), then wait for them to report the results.
+
+## How to run the tests
+*Note: The steps mentioned here have been run and validated on an ubuntu18.04/amd64 machine. If you are running a different os/arch, your steps may differ.*
+
+### One time setup
+
+#### Setting up your local machine
+
+##### Prerequisites
+To run the End-to-End tests, we will be building the required binaries from the code, build container images and push them to a local docker repository. The tests will install the IoT Edge runtime from the binaries on your machine and run the containers using your local repository.
+
+It is important that your machine satisfies the requirements to run IoT Edge. See our installation docs ([Linux](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-linux), [Windows](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-windows)) for more information on prerequisites. In particular, make sure to prepare your machine to access the Microsoft installation pacakges. *The steps to do this are listed below for your convenience. Again, please look at the section in the linked document for your os/arch combination if it is different from ubuntu18.04/amd64*
+~~~sh
+# Setup the repository information
+curl https://packages.microsoft.com/config/ubuntu/18.04/multiarch/prod.list > ./microsoft-prod.list
+
+# Setup the Microsoft GPG pulic key in the apt's trusted list.
+sudo cp ./microsoft-prod.list /etc/apt/sources.list.d/
+curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg
+sudo cp ./microsoft.gpg /etc/apt/trusted.gpg.d/
+
+# Run update to update the package lists on your machine
+sudo apt-get update
+~~~
+
+Also, create a rootCA certificate for your machine *(See [Create Certs](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-create-test-certificates?view=iotedge-2020-11#create-root-ca-certificate) for more details)*. Navigate to the directory where the certificate generation scripts are found (typically iotedge/tools/CACertificates) and run

```

Comment:
Do we just create a root CA cert and that's it for now? Is there another implied step? If it will be used later in the procedure, it would be good to mention that here.